src/axom/quest/SamplingShaper.hpp

```diff
@@ -108,10 +116,29 @@ class InOutSampler
     m_octree->generateIndex();
   }
 
-  void sampleInOutField(mfem::DataCollection* dc,
-                        shaping::QFunctionCollection& inoutQFuncs,
-                        int sampleRes)
+  template <int FromDim, int ToDim>
+  std::enable_if_t<ToDim != DIM, void> sampleInOutField(
+    mfem::DataCollection*,
+    shaping::QFunctionCollection&,
+    int,
+    PointProjector<FromDim, ToDim>)
+  {
+    static_assert(
+      ToDim != DIM,
+      "Do not call this function -- it only exists to appease the compiler!"
+      "Projector's return dimension (ToDim), must match class dimension (DIM)");
+  }
```

Comment:
Note this function can never be called since the dimension of the return point (`ToDim`) has to match that of the spatial index (`DIM`). The compiler complained (a lot) when it was not defined though!